### ISSUE TITLE
Challenge Response not working in AntiX Linux (Debian Stretch)

### ISSUE THREAD

**Issue by cryptopacker** [keepassxreboot/keepassxc#2925]
I have been using KeepassXC with Challenge/Response for quite a while on various Linux distros with no issues, but for some reason, it's not working on AntiX. It's greyed out, so it cannot be checked. I installed the deb package of version 2.3.4 and also tried the AppImage. Both show Challenge Response greyed out with Yubikey inserted. I tried other USB ports but no luck. Any suggestions? Thank you.

**Comment by droidmonkey** [comment]
Does the Yubikey show up when issuing ```lsusb```?

**Comment by cryptopacker** [comment]
Yes, it shows up in [code]lsusb[/code]:
[code] Bus 001 Device 010: ID 1050:0407 Yubico.com[/code]

Thanks for your response and suggestion. I just now installed the Personalization tool and tried testing Challenge-Response using the available tool, but I get "Unknown error occurred" in red in the right-hand pane and below that, all info shows N/A. I notice that's true no matter what green menu item I choose at the top.

**Comment by droidmonkey** [comment]
Unfortunately I think your Yuibkey is not working.

**Comment by cryptopacker** [comment]
Well, it is working perfectly with KeepassXC on my desktop which has MX Linux installed. That's the weird thing. I can't figure it out.

**Comment by cryptopacker** [comment]
This is strange. Following info [here](https://forum.yubico.com/viewtopicd5cb.html?p=8900) I found by running KeepassXC as root, the challenge-response works, but I don't think that's really a solution. So it appears to be a permissions issue. Any idea how I can address this?

**Comment by droidmonkey** [comment]
It appears to be a permissions issue with your usb. This isn't a KeePassXC issue at this point.

**Comment by qwrtln** [comment]
@cryptopacker I struggled with the same problem for a while now. It seems as if I lacked specific udev rules for challenge/response mode, which can be found here: https://github.com/Yubico/yubikey-personalization/blob/master/70-yubikey.rules

Put them in `/etc/udev/rules.d/` directory and then run `sudo udevadm control --reload-rules && sudo udevadm trigger` to reload. KeePassXC should recognize the key without sudo from now on.

**Comment by qwrtln** [comment]
@cryptopacker I struggled with the same problem for a while now. It seems as if I lacked specific udev rules for challenge/response mode, which can be found here: https://github.com/Yubico/yubikey-personalization/blob/master/70-yubikey.rules

Put them in `/etc/udev/rules.d/` directory and then run `sudo udevadm control --reload-rules && sudo udevadm trigger` to reload. KeePassXC should recognize the key without sudo from now on.
